common/calclib/fixed_point_utils.h

```diff
@@ -53,4 +52,19 @@ int64_t int_cast(T&& arg) {
     return fp_cast<int64_t>(std::forward<T>(arg));
 }
 
+template <typename U, typename T>
+void narrow_down(U& lhs, T& rhs) {
+    FIXED_POINT_ASSERT((lhs >= 0) && (rhs >= 0), "narrow_down: wrong args");
```

Comment:
why we need this excessive parentheses? operators precedence already guarantee right calculation order and this part is simple and clearer without parentheses

Reply:
OK, done